src/controllers/check-auth.controller.ts

```diff
@@ -1,28 +1,38 @@
 import { Request, Response } from "express";
 import { verify, decode, JsonWebTokenError, sign, JwtPayload } from 'jsonwebtoken';
 
+export interface ITokenUser {
+  id: string,
+  iat?: number,
+  exp?: number
+}
+
 export const checkAuth = async (req: Request, res: Response) => {
   const { token } = req.params;
-
+  
   if (!token) {
     return res.status(401).send({ succes: false, message: 'Unauthorized: Token not found' });
   }
 
-  let currentUser: string | JwtPayload;
+  let currentUser: ITokenUser;
 
-  verify(token, process.env.JWT_KEY, (err, tokenUser) => {
+  verify(token, process.env.JWT_KEY, (err, tokenUser: ITokenUser) => {
     if (err) {
       return res.status(401).send({ succes: false, message: 'Unauthorized: Bad token' });
     }
-    currentUser = tokenUser;
+    currentUser = { id: tokenUser.id };
```

Comment:
tokenUser has user and iat variables. To reach id, we may have to use tokenUser.user._id. On this approach currentUser is undefined:

tokenUser: {
  user: {
    _id: '641444ccb84980eb625d2fdf',
    fullName: 'New Teser',
    email: 'a@ab.com',
    password: 'cfc678a628ef85d440ba9023452a6cd4622fef416f97299e7a8a3002a4058ca3',
    role: 'patient',
    phoneNumber: '001234',
    __v: 0
  },
  iat: 1679470238
}
currentUser: { id: undefined }

Reply:
the jwt encodes and object of type { id: string }, not an object of type { user: { id: string } }.
From the auth app we get that first object encoded and we send it to the server